### ISSUE TITLE
False positive from indendation rule on a leading comment for a block

### ISSUE THREAD

**Issue by lsg-richard** [sasstools/sass-lint#1209]
The following code triggers a warning about incorrect indentation on the line with a comment:

```sass
.card-item-component
  &:hover
    .card-item-component__icon-menu
      opacity: 1

  // This line triggers the indentation warning
  &__title
    padding: 5px
```

sass-lint version 1.12.1

Output: `error  Expected indentation of 4 spaces but found 2  indentation`

The expected result is that the above code does not trigger an indentation warning.

Note that if the comment is only at one indentation level back from the previous block then the indentation error is not triggered. In the case above the comment is two levels of indentation back from the previous block, and that's what triggers the issue. If the comment is only one level of indentation back then it's fine.

For example, the following does NOT trigger the issue.

```sass
.card-item-component
  &:hover
    opacity: 1

  // This line will not trigger an indentation warning
  &__title
    padding: 5px
```

**Comment by James-Bovis** [comment]
Would love to see this fixed, is a daily pain for me 🙏
